qlkube/src/decorateSubscription.js

```diff
@@ -28,9 +28,79 @@ function checkMetadata(metadataItemA, metadataItemB) {
   );
 }
 
-function checkLabelSelector(subVar, payloadVar) {
-  // TODO: check on LabelSelector
-  return true;
+function splitLabelSelector(labelSelector) {
+  let key;
+  let values = [];
+
+  /*
+   * Split labelSelector into key and values
+   * First case: label selector into the form "crownlabs.polito.it/workspace=sid"
+   * Second case: label selector into the form "crownlabs.polito.it/workspace in (sid)"
+   */
+  if (labelSelector.includes('=')) {
+    let [sel0, sel1] = labelSelector.split('=').map(s => s.trim());
+    key = sel0;
+    values.push(sel1);
+  }
+
+  if (labelSelector.includes(' in ')) {
+    let [sel0, sel1] = labelSelector
+      .replaceAll('(', ' ')
+      .replaceAll(')', ' ')
+      .split(/\s+in\s+/)
+      .map(s => s.trim());
+    key = sel0;
+    values = sel1.split(',').map(v => v.trim());
+  }
+
+  return { key, values };
+}
+
+function checkLabelSelector(subVar, payloadVar, ruid) {
+  let { labels } = payloadVar;
+  let { labelSelector } = subVar;
+  let result = false;
+
+  graphqlLogger(
+    `[i] (${ruid}) Perform check on ${JSON.stringify(
+      labels
+    )} labels with ${labelSelector} label selector`
+  );
+
+  /*
+   * No labelSelector, so all resources are watched
+   */
+  if (!labelSelector) {
+    graphqlLogger(
+      `[i] (${ruid}) The result of the check is true due to empty labelSelector`
+    );
+    return true;
+  }
+  /*
+   * The resource has no labels, so no match
+   */
+  if (!labels) {
+    graphqlLogger(
+      `[i] (${ruid}) The result of the check is false due to empty labels`
+    );
+    return false;
+  }
+
+  if (typeof labels === 'string') labels = JSON.parse(labels);
```

Comment:
It's better to wrap this in a try/catch block, I guess the labels variable could contain forged contents and stuff alike. `JSON.parse` throws an exception if malformed JSON input is read. If you have an external try/catch you don't need to put one here, otherwise the whole process might be killed.

Reply:
To be honest, the `labels` variable is already an object, this is a forgetfulness. I remove this line.